lib/DXIL/DxilPDB.cpp

```diff
@@ -311,7 +317,7 @@ HRESULT hlsl::pdb::WriteDxilPDB(IMalloc *pMalloc, IDxcBlob *pContainer, ArrayRef
   Writer.AddEmptyStream(); // DBI
   Writer.AddEmptyStream(); // IPI
   
-  Writer.AddStream({ (char *)pContainer->GetBufferPointer(), pContainer->GetBufferSize() }); // Actual data block
+  Writer.AddStream({ (char *)ContainerData.data(), ContainerData.size() }); // Actual data block
```

Comment:
This one looks a bit trickier, but I think you can change the `Data` member of `Stream` and the parameter to `AddStream` to const without any consequences. Which would allow this cast to be to `const char *`